### DISCUSSION TITLE
Simplifying property configuration

### DISCUSSION THREAD

**Discussion by ghost** [hedgehogqa/fsharp-hedgehog#342] [Ideas]
I've been skimming over the property module, and I think it needs some TLC. We have no fewer than 8 `report` methods, and I think we can probably do better here. What I was thinking is that we could maybe make combinators that allow configuring a property, something like:

```fsharp
let property = // ...
property
|> Property.withTests 500
|> Property.withRecheck seed size
|> Property.check
```

This would translate into the LINQ interface like:

```csharp
var property = // ...
property
    .WithTests(500)
    .WithRecheck(seed, size)
    .Check();
```

I'm curious to hear thoughts on this. Apologies up front if this was suggested before.

/cc @moodmosaic @TysonMN @dharmaturtle

**Comment by TysonMN** [comment]
This looks great to me.  I think this would be an excellent improvement.  In fact, I think it would make my work on issue #332 / draft PR #336 easier.

Would anyone like to start working on this now?

**Reply by dharmaturtle** [reply]
I'm a bit busy this weekend, but I can look into it the following week - unless someone feels like beating me to the punch.

**Reply by ghost** [reply]
I created issue #343 for this. Probably the simplest way to do this is to make property a `PropertyConfig * Gen<(Journal * Outcome<'a>)>`?
